src/hotspot/share/classfile/classFileParser.cpp

```diff
@@ -5637,8 +5632,9 @@ void ClassFileParser::update_class_name(Symbol* new_class_name) {
 // For an unsafe anonymous class that is in the unnamed package, move it to its host class's
 // package by prepending its host class's package name to its class name and setting
 // its _class_name field.
-void ClassFileParser::prepend_host_package_name(const InstanceKlass* unsafe_anonymous_host, TRAPS) {
-  ResourceMark rm(THREAD);
+void ClassFileParser::prepend_host_package_name(const InstanceKlass* unsafe_anonymous_host) {
+  JavaThread* current = JavaThread::current();
```

Comment:
Can you pass Thread* current as the first parameter instead?  We're trying to minimize materializing the current thread.

Reply:
How about if "ResourceMark rm(THREAD)" is changed to "ResourceMark rm;" ?  Then no Thread* is need in the function.

Reply:
This seems fine to me.

Reply:
I think the primary goal should be not avoid calling Thread::current() as much as possible.  Passing Thread* thread around, even as the first parameter just for a ResourceMark seems like unnecessary typing to me.  Getting the current thread isn't expensive.  From experience, if we optimized all the Thread::currents away, we'd get no benefits to performance, and it's nice that it's hidden here.  If we have an impasse here, go ahead and add the Thread parameter at the beginning, but I'd rather not see this extra parameter.

Some previous project that I did (I won't admit how long ago that was), was to go through all the sources in search of Thread::current() and use whatever current thread was passed in, usually via TRAPS. I didn't gain any performance from this but the code looked nicer in the end.